### PR TITLE
[5.0] Add event for session started

### PR DESCRIPTION
**Pull request by ghost**
resolves laravel/framework#7906

### CODE REVIEW

File: src/Illuminate/Session/composer.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
         "php": ">=5.4.0",
         "illuminate/contracts": "5.0.*",
         "illuminate/support": "5.0.*",
+        "illuminate/events" : "5.0.*",
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/composer.json]: Please put this into the suggest list.

File: src/Illuminate/Session/Middleware/StartSession.php [hunk 2]
```diff
@@ -102,6 +102,8 @@ protected function startSession(Request $request)
 
 		$session->start();
 
+		app('events')->fire('session.started');
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/Middleware/StartSession.php]: You can't use that helper here. You must inject an implementation of the events contract in the constructor.

**Comment by taylorotwell** [comment]
There is no session because there is no HTTP request. People need to be more in the habit of creating middlewares for this kind of thing. There is no bug to resolve.

**Comment by swiggles** [comment]
Ok thanks for that, will look into if this can be achieved another way. 

Maybe I can fire an event in some middleware put it lower down in the kernels middleware array that the start session middleware. I can then listen for that event in the service providers. That should do it.

**Comment by garygreen** [comment]
@taylorotwell In packages that need to access the session data in service providers, how should this be done now with middleware? Do you add in your guide for installing the package to add a new 'Middleware' to the array?

**Comment by ghunti** [comment]
Hi all.
I also have a package that mimics the behavior of the "errors" variable to inject a "messages" variable on all views.
Of course I can provide a new middleware with the package and ask users to add it to their Kernel array, but I would like to see a cleaner approach.
Imagine that if someone else is creating a package that depends on my own package, they will have to also instruct their users to add my package's middleware to their Kernel. With the growing number of packages this could rapidly become cumbersome.
Just my 2 cents.
Thanks

**Comment by gilbertoalbino** [comment]
As I said on StackOverflow, sometimes people seem to forget Laravel is built upon PHP, and completely forget how to use PHP on its behalf.

https://stackoverflow.com/questions/34577946/how-to-retrieve-session-data-in-service-providers-in-laravel/64821595#64821595

**Comment by driesvints** [comment]
@gilbertoalbino this PR is 5 years old.

**Comment by gilbertoalbino** [comment]
@driesvints So what? PHP is 25 years old and a lot of things remain the same!
People try to do "everything" in Laravel, as if Laravel was a new programming language, and simple things get so complicated by PHP being ignored in Laravel Projects.

**Pull request closed without merging**